### PR TITLE
WIKI-1408: Users can remove attached files on wiki page even if they don't have edit permission

### PR DESCRIPTION
**Pull request by AzmiTouil**
The problem that all users can delete an attachment in a wiki page even if they don't have the right to edit the page.

### CODE REVIEW

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
 
+  private WikiService wikiService;
+
   public RemoveAttachmentPermissionFilter() {
     this(null);
   }
 
   public RemoveAttachmentPermissionFilter(String messageKey) {
     super(messageKey, UIExtensionFilterType.MANDATORY);
+    this.wikiService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(WikiService.class);
```
Comment (boubaker) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: Please use CommonsUtils.getService()

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
+
   }
 
   @Override
   public boolean accept(Map<String, Object> context) throws Exception {
-    // TODO there is no permissions on attachments, why do we need a filter ?
-    return true;
+    WikiPageParams pageParams = Utils.getCurrentWikiPageParams();
```
Comment (boubaker) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: Please test on page params nullability, if null return false.
Same for owner, page name, page type.

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
+    Page page = wikiService.getPageOfWikiByName(pageParams.getType(), pageParams.getOwner(), pageParams.getPageName());
+    if (page == null) {
+      return false;
+    }
+    return wikiService.hasPermissionOnPage(page, PermissionType.EDITPAGE, ConversationState.getCurrent().getIdentity());
```
Comment (boubaker) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: you may have a null on `ConversationState.getCurrent()`  or `ConversationState.getCurrent().getIdentity()`

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java (hunk 1 above)
Comment (AzmiTouil) [reply wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: Okay

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java (hunk 2 above)
Comment (AzmiTouil) [reply wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: done

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java (hunk 3 above)
Comment (AzmiTouil) [reply wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/RemoveAttachmentPermissionFilter.java]: I added nullability tests .

**Pull request merged**